File: Back/NodeJS/modules/users/tests/unit/implems/MockUserRepository.ts
import { IUserRepository } from "../../../src/domain/users/IUserRepository";
import { User } from "../../../src/domain/users/User";

export class MockUserRepository implements IUserRepository {
    constructor(user?: User) {
        this.User = user;
    }
    IsExists(email: string, userId: string, username: string): Promise<boolean> {
        if(this.User) {
            return Promise.resolve(true);
        }
        return Promise.resolve(false);
    }
    Save(user: User): Promise<void> {
        this.User = user;
        return Promise.resolve();
    }
    User?: User;

}

File: Back/.Net/Modules/Users/BetFriends.Users.UnitTests/Implems/MockUserRepository.cs
using BetFriends.Users.Domain.Users;

namespace BetFriends.Users.UnitTests.Implems;

internal class MockUserRepository(User user = null!) : IUserRepository
{
    public User User { get; private set; } = user;

    public Task<bool> IsUserExistAsync(string username, string email)
    {
        if(User != null)
            return Task.FromResult(true);
        return Task.FromResult(false);
    }

    public Task SaveAsync(User user)
    {
        User = user;
        return Task.CompletedTask;
    }
}